packages/react/src/components/List/HierarchyList/HierarchyList.jsx

```diff
@@ -359,9 +359,12 @@ const HierarchyList = ({
    * the total filtered array. The next category's children then needs to
    * be searched in the same fashion.
    * @param {String} text keyed values from search input
+   * @param {Array} currentItems the current state of items.
+   *                             Used to maintain state when the list is updated (drag and drop)
+   *                             and there's a current search value
    */
-  const handleSearch = (text) => {
-    const tempFilteredItems = searchForNestedItemValues(items, text);
+  const handleSearch = (text, currentItems = null) => {
```

Comment:
Nit: Wouldn't it make more sense that the signature had a required  "haystack" parameter? Now it can be either implicit or explicit.

Reply:
sure. updated.